src/Plugin/rest/resource/YouTubeRestResource.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\youtube_proxy\Plugin\rest\resource;
+
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\rest\ResourceResponse;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "youtube_rest_resource",
+ *   label = @Translation("YouTube rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/api/yt_proxy/{video_id}"
+ *   }
+ * )
+ */
+class YouTubeRestResource extends ResourceBase {
+
+  /**
+   * A current user instance.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+    protected $currentUser;
+    private $_cacheId;
+
+  
+  /**
+   * {@inheritdoc}
+   */
+    public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+        $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+        $instance->logger = $container->get('logger.factory')->get('youtube_proxy');
+        $instance->currentUser = $container->get('current_user');
+        $instance->currentRequest = $container->get('request_stack')->getCurrentRequest();
+        return $instance;
+    }
+
+    /**
+     * Responds to GET requests.
+     *
+     * @param string $payload
+     *
+     * @return \Drupal\rest\ResourceResponse
+     *   The HTTP response object.
+     *
+     * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+     *   Throws exception expected.
+     */
+    public function get($video_id = null) {
+        // You must to implement the logic of your REST Resource here.
+        // Use current user after pass authentication to validate access.
+        if (!$this->currentUser->hasPermission('access content')) {
+            throw new AccessDeniedHttpException();
+        }
+
+        // Check if the video ID is set.
+        if ($video_id == null || !isset($video_id)) {
```

Comment:
`$video_id == null || !isset($video_id)` kan ook `!empty($video_id)` zijn. Is iets netter.

Reply:
Yes ik ga dit aanpassen